include/cuco/distinct_count_estimator.cuh

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cuco/cuda_stream_ref.hpp>
+#include <cuco/detail/hyperloglog/hyperloglog.cuh>
+#include <cuco/distinct_count_estimator_ref.cuh>
+#include <cuco/hash_functions.cuh>
+#include <cuco/utility/allocator.hpp>
+#include <cuco/utility/cuda_thread_scope.cuh>
+
+#include <cstddef>
+#include <iterator>
+#include <memory>
+
+namespace cuco {
+template <class T,
+          int32_t Precision        = 11,
```

Comment:
I would say yes because this is the one parameter that allows you to trade speed/memory for accuracy. For cudf we should fix it to 11 or 13 (something that still fits in shmem. The problem I'm having with exposing this as a non-type tparam is that now CTAD no longer works -.-